### PR TITLE
update blockbuffer code

### PR DESCRIPTION
**Pull request by visakhsuku**

### CODE REVIEW

File: design/buffer.html [hunk 1]
```diff
@@ -706,15 +706,19 @@ <h2>Class Block Buffer </h2>
...
 
 public:
-	BlockBuffer(int block_num); 
-    int getblocknum();
-	struct HeadInfo getheader();
-	void setheader(struct HeadInfo head); 
-	virtual ~BlockBuffer();
+	int getBlockNum();
+	struct HeadInfo getHeader();
+	void setHeader(struct HeadInfo head);
```
Comment (appu313) [inline design/buffer.html]: make the return type and argument type as pointer

File: design/buffer.html [hunk 2]
```diff
@@ -706,15 +706,19 @@ <h2>Class Block Buffer </h2>
...
+	int getBlockType(int bufferIndex);
```
Comment (appu313) [inline design/buffer.html]: no need for bufferIndex. Calculate that in the body of the function

File: design/buffer.html (hunk 1 above)
Comment (visakhsuku) [reply design/buffer.html]: updated

File: design/buffer.html (hunk 2 above)
Comment (visakhsuku) [reply design/buffer.html]: updated

File: design/buffer.html [hunk 3]
```diff
@@ -706,15 +706,19 @@ <h2>Class Block Buffer </h2>
...
+	void setBlockType(int blockType);
+	BlockBuffer(int blockNum);
+	BlockBuffer(char blockType);
 };</code></pre> 
                             </div>
                             <p>The following is the specification of methods of BlockBuffer</p>
```
Comment (appu313) [inline design/buffer.html]: constructor 1

File: design/buffer.html [hunk 4]
```diff
@@ -725,46 +729,55 @@ <h2>BlockBuffer (Constructor)</h2>
...
+                                                <td>blockNum</td>
                                                 <td>int</td>
                                                 <td>block number of the corresponding block</td>
                                             </table>
                                         </div>
                                         <h5>Return type:</h5>
                                         <p>void</p>
                                         <h5>Description:</h5>
-                                        <p></p>
+                                        <p>Should be called if the block already present in the disc.</p>
```
Comment (appu313) [inline design/buffer.html]: "to be called when the block already exists on the disk"

File: design/buffer.html [hunk 5]
```diff
@@ -725,46 +729,55 @@ <h2>BlockBuffer (Constructor)</h2>
...
                                         <div class="table-responsive">
                                             <table class="table table-bordered table-hover">
-                                                <td>blk_no</td>
+                                                <td>blockNum</td>
                                                 <td>int</td>
                                                 <td>block number of the corresponding block</td>
                                             </table>
                                         </div>
                                         <h5>Return type:</h5>
                                         <p>void</p>
```
Comment (appu313) [inline design/buffer.html]: return type is nil

File: design/buffer.html [hunk 6]
```diff
@@ -725,46 +729,55 @@ <h2>BlockBuffer (Constructor)</h2>
...
-                                        <p></p>
+                                        <p>Should be called if the block already present in the disc.</p>
                                         <h5><a data-toggle="collapse" href="#algo-blkbuffer-constructor"><span class="fas fa-code"></span>Algorithm:</a></h5>
                                         <div id="algo-blkbuffer-constructor" class="collapse">
                                             <div class="code">
-                                                <pre><code>Blockbuffer::BlockBuffer(int blk_no){
-    // set the block_num field of this object to blk_no;
+                                                <pre><code>Blockbuffer::BlockBuffer(int blockNum){
+    				this.blockNum = blockNum;
+				getBlock();
```
Comment (appu313) [inline design/buffer.html]: 1. set the blockNum field of the object to input argument.
2. call getBlock()  to copy the block into buffer.

File: design/buffer.html [hunk 7]
```diff
@@ -706,65 +706,78 @@ <h2>Class Block Buffer </h2>
...
-                                        <p>Nil</p>                            
+                                        <div class="table-responsive">
+                                            <table class="table table-bordered table-hover">
+                                                <td>blockType</td>
+                                                <td>char</td>
+                                                <td>block type of the corresponding block</td>
+                                            </table>
+                                        </div>
                                         <h5>Return type:</h5>
                                         <p>void</p>
```
Comment (appu313) [inline design/buffer.html]: return type Nil

File: design/buffer.html [hunk 8]
```diff
@@ -706,65 +706,78 @@ <h2>Class Block Buffer </h2>
...
                                         <h5>Description:</h5>
-                                        <p></p>
-                                        <h5><a data-toggle="collapse" href="#algo-blkbuffer-destructor"><span class="fas fa-code"></span>Algorithm:</a></h5>
-                                        <div id="algo-blkbuffer-destructor" class="collapse">
+                                        <p>Should be called if a new block has to be allocated in the disc.</p>
```
Comment (appu313) [inline design/buffer.html]: to be called if a new block is to be allocated in the disc.

File: design/buffer.html [hunk 9]
```diff
@@ -706,65 +706,78 @@ <h2>Class Block Buffer </h2>
...
+                                        <h5><a data-toggle="collapse" href="#algo-blkbuffer-constructor1"><span class="fas fa-code"></span>Algorithm:</a></h5>
+                                        <div id="algo-blkbuffer-constructor1" class="collapse">
                                             <div class="code">
-                                                <pre><code>BlockBuffer::~BlockBuffer(){
-    //release the block corresponding to block_no from buffer calling <a href="#releaseblk">releaseBlock</a>() method of buffer class.
+                                                <pre><code>Blockbuffer::BlockBuffer(char blockType){
+    					this.blockNum = getFreeBlock(blockType);
```
Comment (appu313) [inline design/buffer.html]: // allocate a block on the disc and a buffer in memory to hold the new block of given type using getFreeBlock function.

// set the blockNum field of the object to the allocated block number.

**Pull request merged**